### PR TITLE
JAMES-3928 ReadOnlyUsersRepository: Master slave set up

### PR DESCRIPTION
**Pull request by quantranhong1999**
TODO:
- See if the LDAP supports different SSL settings for distinct URI, otherwise, we would need to assume the setting are the same.
- Other tests?
- Additional work upon review
- Docs & sample configuration

### CODE REVIEW

**Comment by chibenwa** [comment]
> See if the LDAP supports different SSL settings for distinct URI, otherwise, we would need to assume the setting are the same.

Same settings seems fair to me.

File: server/data/data-ldap/src/main/java/org/apache/james/user/ldap/LdapRepositoryConfiguration.java [hunk 1]
```diff
@@ -143,7 +145,8 @@ public static Builder builder() {
     }
 
     public static LdapRepositoryConfiguration from(HierarchicalConfiguration<ImmutableNode> configuration) throws ConfigurationException {
-        String ldapHost = configuration.getString("[@ldapHost]", "");
+        List<String> ldapHosts = Splitter.on(",")
```
Comment (chibenwa) [inline server/data/data-ldap/src/main/java/org/apache/james/user/ldap/LdapRepositoryConfiguration.java:148]: trim

And create a new property, `ldapHosts`, falling back to `ldapHost`.

**Review by chibenwa — COMMENTED** [review]
The test suite looks REALLY solid to me!

**Review by Arsnael — COMMENTED** [review]
Read it, nothing to add :)

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

File: server/data/data-ldap/src/main/java/org/apache/james/user/ldap/LdapRepositoryConfiguration.java [hunk 2]
```diff
@@ -202,13 +207,14 @@ public static LdapRepositoryConfiguration from(HierarchicalConfiguration<Immutab
...
      * Note that users are actually authenticated by binding against the LDAP
-     * server using the users &quot;dn&quot; and &quot;credentials&quot;.The
+     * servers using the users &quot;dn&quot; and &quot;credentials&quot;.The
      * value of this field is taken from the value of the configuration
-     * attribute &quot;ldapHost&quot;.
+     * attribute &quot;ldapHosts&quot; and fallback to the legacy attribute &quot;ldapHost&quot;.
+     * URLs are split by the comma  &quot;,&quot; character.
      */
-    private final String ldapHost;
+    private final List<String> ldapHosts;
```
Comment (vttranlina) [inline server/data/data-ldap/src/main/java/org/apache/james/user/ldap/LdapRepositoryConfiguration.java:217]: or `List<URI> ldapHosts`

**Review by vttranlina — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**